test/common/stats/stats_impl_test.cc

```diff
@@ -478,5 +478,14 @@ TEST(RawStatDataTest, Truncate) {
   alloc.free(*stat);
 }
 
+TEST(RawStatDataTest, HeapAlloc) {
+  HeapRawStatDataAllocator alloc;
+  RawStatData* stat_1 = alloc.alloc("ref_name");
+  RawStatData* stat_2 = alloc.alloc("ref_name");
```

Comment:
maybe also do stat_3 with "not_ref_name" and check it's not equal, and that none of these are nullptr.